src/components/Modal/Modal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useRef } from 'react'
+import PropTypes from 'prop-types'
+import { Portal } from '../Portal'
+import styles from './Modal.module.scss'
+import useOutsideClick from '../../hooks/a11y/useOutsideClick'
+import useBodyScrollLock from '../../hooks/a11y/useBodyScrollLock'
+import useTrapFocus from '../../hooks/a11y/useTrapFocus'
+import useHideAriaSiblings from '../../hooks/a11y/useHideAriaSiblings'
+
+/**
+ * Component renders the internal content of a modal
+ *
+ * @param {object} props - the component's props
+ * @param {React.ReactChildren} props.children - the component's children
+ * @param {(val: boolean) => any} props.toggle - callback function that toggles the
+ * show modal state
+ *
+ * @return {JSX.Element}
+ */
+const ModalContent = ({
+  ariaDescribedBy,
+  ariaLabelledBy,
+  children,
+  toggle,
+  showModal,
+}) => {
+  const modalRef = useRef(null)
+
+  // Hooks! All the thingggs!!!!!
+  useOutsideClick(modalRef, () => toggle(false))
```

Comment:
This could be an `onDismiss` or something since it only ever "toggles" off.

Reply:
I agree with that –  looking at the state of the component now, we the modal does only set the toggle state to `false`.

Reply:
Yeah, disregard my other comment I like `onDismiss` :)